packages/calcite-components/src/components/sheet/sheet.e2e.ts

```diff
@@ -541,4 +541,144 @@ describe("calcite-sheet properties", () => {
+          style="width: 200px"
+          id="combobox"
+        >
+          <calcite-combobox-item value="Grand 1" text-label="Grand 1">
+            <calcite-combobox-item value="Parent 1" text-label="Parent 1">
+              <calcite-combobox-item value="Child 1" text-label="Child 1"></calcite-combobox-item>
+              <calcite-combobox-item value="Child 2" text-label="Child 2"></calcite-combobox-item>
+            </calcite-combobox-item>
+          </calcite-combobox-item>
+        </calcite-combobox>
+        <calcite-dropdown scale="s" width-scale="s" id="dropdown">
+          <calcite-button icon-end="hamburger" appearance="outline" slot="trigger">Scale S</calcite-button>
+          <calcite-dropdown-group group-title="View">
+            <calcite-dropdown-item icon-end="list-bullet" selected>List</calcite-dropdown-item>
+            <calcite-dropdown-item icon-end="grid">Grid</calcite-dropdown-item>
+          </calcite-dropdown-group>
+        </calcite-dropdown>
+        <calcite-popover
+          heading="Heading"
+          label="right end popover"
+          reference-element="popover-button"
+          placement="right-end"
+          id="popover-heading"
+          closable
+          style="width: 25vw"
+          id="popover"
+        >
+          <div style="padding: 0.5rem 1rem 0.75rem">
+            <p style="margin-top: 0">Example Popover.</p>
+          </div>
+        </calcite-popover>
+        <calcite-button appearance="outline" id="popover-button" icon-start="popup">Example Popover</calcite-button>
+        <calcite-tooltip placement="auto" reference-element="tooltip-auto-ref"> Example Tooltip </calcite-tooltip>
+        <calcite-button appearance="outline" id="tooltip-auto-ref">auto</calcite-button>
+      </div>
+    </calcite-modal>
+    <calcite-button onClick="openComponent('example-sheet')"> Open Sheet </calcite-button>
+  `;
+
+  it("closes a stack of open components sequentially in visual order", async () => {
+    const page = await newE2EPage();
+    await page.setContent(componentStack);
+    await skipAnimations(page);
+
+    async function openAndCheckVisibility(page: E2EPage, element: E2EElement) {
+      element.setProperty("open", true);
+      await page.waitForChanges();
+      expect(await element.isVisible()).toBe(true);
+    }
+
+    const sheet = await page.find("calcite-sheet");
+    await openAndCheckVisibility(page, sheet);
+
+    const firstModal = await page.find("#example-modal");
+    await openAndCheckVisibility(page, firstModal);
+
+    const secondModal = await page.find("#another-modal");
+    await openAndCheckVisibility(page, secondModal);
+
+    async function testInputPicker(page: E2EPage, pickerSelector: string, modal: E2EElement) {
+      const inputPicker = await page.find(pickerSelector);
+      inputPicker.click();
+      await page.waitForChanges();
+      expect(await inputPicker.getProperty("open")).toBe(true);
+
+      await page.keyboard.press("Escape");
```

Comment:
I think each escape should check to make sure only the last opened component is closed. Basically have a stack of all the components that are opened and verify only the expected one is closed at a time. Each call should close the next one. Currently, its only checking one component at a time.

Reply:
`calcite-input-date-picker` and `time-picker` don't get stacked, so these 2 will have their own stacks to be checked.

Reply:
We can probably remove input-date-picker and time-picker from the tests. We don't need to test components that don't focus trap.

Reply:
Maybe there is some benefit in making sure the top component doesn't close focus-trap stack?